crates/red_knot_python_semantic/src/site_packages.rs

```diff
@@ -205,12 +211,12 @@ System site-packages will not be used for module resolution.",
 
 #[derive(Debug, thiserror::Error)]
 pub(crate) enum SitePackagesDiscoveryError {
-    #[error("Invalid --python argument: `{0}` could not be canonicalized")]
-    VenvDirCanonicalizationError(SystemPathBuf, #[source] io::Error),
-    #[error("Invalid --python argument: `{0}` does not point to a directory on disk")]
-    VenvDirIsNotADirectory(SystemPathBuf),
-    #[error("--python points to a broken venv with no pyvenv.cfg file")]
-    NoPyvenvCfgFile(#[source] io::Error),
+    #[error("Invalid {1}: `{0}` could not be canonicalized")]
+    VenvDirCanonicalizationError(SystemPathBuf, SysPrefixPathOrigin, #[source] io::Error),
+    #[error("Invalid {1}: `{0}` does not point to a directory on disk")]
+    VenvDirIsNotADirectory(SystemPathBuf, SysPrefixPathOrigin),
+    #[error("{0}: points to a broken venv with no pyvenv.cfg file")]
```

Comment:
I think we should get rid of the colon here. I manually tested this by:
1. Creating a virtual environment using `uv venv`
2. Breaking the virtual environment by running `rm .venv/pyvenv.cfg`
3. Activating the virtual environment by running `source .venv/bin/activate`
4. Running `../ruff/target/debug/red_knot check`

The error message displayed to the terminal was:

```
~/dev/typeshed-stats (main) % uv run ../ruff/target/debug/red_knot check
Red Knot failed
  Cause: Invalid search path settings
  Cause: Failed to discover the site-packages directory: `VIRTUAL_ENV` environment variable: points to a broken venv with no pyvenv.cfg file
```

And I don't think the second colon on the last line there is necessary

```suggestion
    #[error("{0} points to a broken venv with no pyvenv.cfg file")]
```